src/client/opamAdminCommand.ml

```diff
@@ -1005,13 +1005,43 @@ let add_constraint_command =
         let deps =
           OpamFormula.map (function
               | (n,c as atom) ->
-                if n = name then Atom (n, (add_cstr nv n c))
+                if n = name then Atom (n, (add_cstr OpamFormula.ands cstr nv n c))
                 else Atom atom)
             deps0
         in
-        if deps <> deps0 then
+        let depopts0 = OpamFile.OPAM.depopts opam in
+        let conflicts0 = OpamFile.OPAM.conflicts opam in
+        let contains name =
+          OpamFormula.fold_left (fun contains (n,_) ->
+              contains || n = name) false
+        in
+        let conflicts =
+          if contains name depopts0 then
+            match cstr_opt with
+            | Some (relop, v) ->
+              let icstr =
+                OpamFormula.Atom
+                  (Constraint (OpamFormula.neg_relop relop,
+                               FString (OpamPackage.Version.to_string v)))
+              in
+              if contains name conflicts0 then
+                OpamFormula.map (function
+                    | (n,c as atom) ->
+                      if n = name then Atom (n, (add_cstr OpamFormula.ors icstr nv n c))
+                      else Atom atom)
+                  conflicts0
+              else
+                OpamFormula.ors [ conflicts0; Atom (name, icstr) ]
+            | None ->
+              OpamFormula.map_formula (function
+                  | Atom (n,_) as f -> if n = name then Empty else f
+                  | f -> f) conflicts0
```

Comment:
```suggestion
            | None -> conflicts0
```
As per the manpage: 
"If no version constraint is specified, the command will just simplify existing version constraints on dependencies to the named package."
So no reason to remove the conflict in this case.